tests/test_apply.py

```diff
@@ -91,51 +91,11 @@ def mimic_execution_for_apply(device):
 class TestApplyPureState:
     """Test application of PennyLane operations on the pure state simulator."""
 
-    def test_basis_state(self, shots, tol):
+    @pytest.mark.parametrize("state", [[1, 0, 0, 0], [0, 0, 1, 0]])
+    def test_basis_state(self, shots, tol, state):
         """Test basis state initialization"""
         dev = SimulatorDevice(4, shots=shots)
-        state = np.array([0, 0, 1, 0])
-
-        with mimic_execution_for_apply(dev):
-            dev.apply([qml.BasisState(state, wires=[0, 1, 2, 3])])
-
-        res = dev._state
-
-        expected = np.zeros([2**4])
-        expected[np.ravel_multi_index(state, [2] * 4)] = 1
-        assert np.allclose(res, expected, **tol)
-
-    @pytest.mark.parametrize(
-        "state",
-        [
-            np.array([0, 0]),
-            np.array([1, 0]),
-            np.array([0, 1]),
-            np.array([1, 1]),
-        ],
-    )
-    @pytest.mark.parametrize("device_wires", [3, 4, 5])
-    @pytest.mark.parametrize("op_wires", [[0, 1], [1, 0], [2, 0]])
-    def test_basis_state_on_wires_subset(self, state, device_wires, op_wires, tol):
```

Comment:
Why did you remove this test?

Reply:
It would be nice to check that the behaviour hasn't changed

Reply:
It tests that BasisState parameters are expanded by `_expand_state` in this plugin, but it's not done in this plugin anymore! It's in `BasisState.state_vector()` which is in pennylane itself. This test would be double-coverage which is fine but imo unnecessary

Reply:
Did you at least test it locally? I wanted to make sure that the deleted implementation here was exactly the same (covered the exact same corner cases) than the implementation in `BasisState.state_vector()`

Reply:
I did indeed run the tests before removing them, and I just triple-checked locally by ensuring the `_expand_state`, `dev._state` and `BasisState.state_prep()` all match each other on master. I could copy some test cases to [test_BasisState_state_vector](https://github.com/PennyLaneAI/pennylane/blob/023e001cd437e5283a4ffa1b2f73bda38ebf4bc9/tests/ops/qubit/test_state_prep.py#L167) as well if we're worried about coverage?

Reply:
I believe copying the tests over to PL is the safest choice 👍

Reply:
moved them over in the linked PR below, lmk if we're good to go!

Reply:
Good to go! 🔥